### PR TITLE
Add UFS version validation task

### PR DESCRIPTION
**Pull request by ZacBlanco**
This new task helps expose some more information about available UFS versions to users. The cases for configuration and corresponding outputs are defined below.

- UFS version not configured: skip validation
- UFS version configured, but doesn't exist in list of valid versions: error with info and list of available versions
- UFS version configured, but list of versions empty: this indicates the UFS doesn't support versioning (i.e. non-HDFS), and directs the user to change configuration
- UFS version configured and exists in available version list: success

### CODE REVIEW

**Comment by yuzhu** [comment]
@ZacBlanco license header?

File: integration/tools/validation/src/main/java/alluxio/cli/UfsVersionValidationTask.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    List<String> availableVersions =
+        UnderFileSystemFactoryRegistry.getSupportedVersions(mUfsPath, ufsConf);
+    ValidationTaskResult result = new ValidationTaskResult();
+    result.setName(getName());
+    result.setDesc("Validates that the configured UFS version exists as a library on the "
+        + "system.");
+
+    if (!mConf.isSetByUser(PropertyKey.UNDERFS_VERSION)) {
+      result.setState(ValidationUtils.State.SKIPPED);
+      result.setOutput("The UFS version was not configured by the user.");
```
Comment (jiacheliu3) [inline integration/tools/validation/src/main/java/alluxio/cli/UfsVersionValidationTask.java:46]: Do you think it's possible there's a case where the user should have configured the `UNDERFS_VERSION` but did not?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
@ZacBlanco Thanks! I left a question.

File: integration/tools/validation/src/main/java/alluxio/cli/UfsVersionValidationTask.java (hunk 1 above)
Comment (ZacBlanco) [reply integration/tools/validation/src/main/java/alluxio/cli/UfsVersionValidationTask.java:57]: Users should only configure UFS version in the case that UFSes support a version. In our current implementation, Alluxio doesn't require a version to be specified, even if there is more than one version supported. It just picks a default if it is not configured. So, it is not necessary for a ever configure a version for a UFS to be chosen.

However, there is a a chance that the default version used is _not_ compatible with the particular UFS. This may induce runtime errors upon connection or data transfer. In those cases we defer to the other UFS checks to flesh our those errors. This check is designed just to make sure that the configured version corresponds to a valid Alluxio UFS library. There's no good way to detect is the UFS is _actually_ compatible at runtime. And even if it's not, any errors that occur likely won't say anything along the line of "hey, you're using the wrong client version". It will be much more difficult to decipher.

tl;dr yes, it is possible but determining the case in which the UFS version is wrong is a non-trivial problem

Comment (jiacheliu3) [reply integration/tools/validation/src/main/java/alluxio/cli/UfsVersionValidationTask.java:57]: I see, that's fair to say. (Should tldr be on the top?)

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the improvement!

**Review by yuzhu — APPROVED** [review]
LGTM

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by yuzhu** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please
